### ISSUE TITLE
Enable IAsyncRequestChannel

### ISSUE THREAD

**Issue by roncain** [dotnet/wcf#1494]
We introduced IAsyncRequestChannel internally to use Task-based API's rather than BeginRequest/EndRequest.  But no caller is yet aware to use this new API, and it is not plumbed through all the RequestChannel implementations.

To complete this task, modify ServiceChannel to support IAsyncRequestChannel, and modify all the IRequestChannel implementations to use it.

**Comment by roncain** [comment]
I recommend this work be done after #1257 is resolved because its implementation introduces several request channels that will require modification.

**Comment by zhenlan** [comment]
This will simplify the internal programming model and reduce the code size. The perf impact is to be determined. This is estimated ~1 week work.

**Comment by roncain** [comment]
Currently in progress in the ws-trust feature branch.  Using that branch because its largest impact is across the security layers.

**Comment by roncain** [comment]
The ws-trust feature branch now has working async request channel tests.  But based on what I've learned, we need to discuss how to proceed.  There is a large body of Begin/End async usage throughout many of the IdentityModel and ServiceModel types related to ws-trust.  Rewriting them to be task-based would take probably 3-4 weeks, minimum, and would then be very different from the full framework.  The alternative is porting the Begin/End AsyncResult code from the full framework.  So far, I have just ported APM for the simplest ws-trust test case and wrapped Tasks around it.
